From 37720c38d6c617439df0a13e9396fcb26345dadd Mon Sep 17 00:00:00 2001
From: iglocska <andras.iklody@gmail.com>
Date: Fri, 23 Mar 2018 11:44:47 +0100
Subject: [PATCH] fix: critical API integrity bug, potentially allowing users
 to delete attributes of other events

- a crafted edit for an event (without attribute UUIDs but attribute IDs set) could overwrite an existing attribute
---
 app/Model/Attribute.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/app/Model/Attribute.php b/app/Model/Attribute.php
index b1be7a5148..94ea0d387c 100644
--- a/app/Model/Attribute.php
+++ b/app/Model/Attribute.php
@@ -3128,6 +3128,7 @@ public function editAttribute($attribute, $eventId, $user, $objectId, $log = fal
 			$attribute['data'] = $result['data'];
 			$attribute['value'] = $attribute['value'] . '|' . $result['md5'];
 		}
+		unset($attribute['id']);
 		if (isset($attribute['uuid'])) {
 			$existingAttribute = $this->find('first', array(
 				'conditions' => array('Attribute.uuid' => $attribute['uuid']),
